mfr/providers/osf/provider.py

```diff
@@ -96,6 +99,12 @@ def __init__(self, request, url):
         # }}
 
         name, ext = os.path.splitext(metadata['data']['name'])
+
+        if MAX_FILE_SIZE_TO_RENDER.get(ext) and metadata['data']['size'] > MAX_FILE_SIZE_TO_RENDER.get(ext):
```

Comment:
I believe some providers may have  `metadata['data']['size'] == None`, which would probably throw a type error here.  If size is unknown, let's skip this check.